### PR TITLE
MBL-13195 - add submissions set to assignment to so observer role can get multiple students submissions on assignments

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-13195
affects: Parent
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1006869&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|0
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-13195](https://instructure.atlassian.net/browse/MBL-13195)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.19% | 93.19% | -0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Models/Assignment.swift [hunk 1]
```diff
@@ -114,12 +137,13 @@ extension Assignment {
...
-                submission = sub
-            } else if let submission = submission {
-                client.delete(submission)
-                self.submission = nil
+            if let values = item.submission?.values, values.count > 0 {
+                self.submissions = Set<Submission>()
+                let newSubs = values.map { Submission.save($0, in: client) }
+                self.submissions = Set<Submission>(newSubs)
+            } else if let submissions = submissions {
+                client.delete(Array(submissions))
```
Comment (armstrongnate) [inline Core/Core/Models/Assignment.swift]: Is this needed with `deletionRule="Cascade"`?

**Comment by armstrongnate** [comment]
I installed the student build to check assignment submissions and it appears to be fine.

File: Core/Core/Models/Assignment.swift (hunk 1 above)
Comment (bluwave) [reply Core/Core/Models/Assignment.swift]: yeah, there is a test for that and it was not passing w/ out this line.  I do have it set to deletionRule = Cascade

**Pull request merged**